File: unit_04/w10_d1/lectures/calculatorChall/js/spec/Calculator-spec.js
const Calculator = require('../Calculator');

describe('Calculator', function() {
  let calculator;

  beforeEach(() => {
    calculator = new Calculator();
  });

  describe('#add', () => {
    it('adds two numbers', () => {
      const actual = calculator.add(2, 3);
      const expectation = 5;
      expect(actual).toEqual(expectation);
    });
  });

  describe('#subtract', () => {
    it('subtract two numbers', () => {
      const actual = calculator.subtract(2, 3);
      const expectation = -1;
      expect(actual).toEqual(expectation);
    });
  });

  describe('#multiply', () => {
    it('multiply two numbers', () => {
      const actual = calculator.multiply(2, 3);
      const expectation = 6;
      expect(actual).toEqual(expectation);
    });
  });

  describe('#divide', () => {
    it('divide two numbers', () => {
      const actual = calculator.divide(6, 3);
      const expectation = 2;
      expect(actual).toEqual(expectation);
    });
  });

  describe('with previous answer of 5 found', () => {
    beforeEach(() => {
      calculator.add(2, 3);
    });

    describe('#add', () => {
      it('adds a number to previous answer', () => {
        expect(calculator.add(2)).toEqual(5 + 2);
        expect(calculator.add(2)).toEqual(5 + 2 + 2);
      });
    });

    describe('#subtract', () => {
      it('subtracts a number to previous answer', () => {
        expect(calculator.subtract(2)).toEqual(5 - 2);
        expect(calculator.subtract(2)).toEqual(5 - 2 - 2);
      });
    });

    describe('#multiply', () => {
      it('multiplys a number to previous answer', () => {
        expect(calculator.multiply(2)).toEqual(5 * 2);
        expect(calculator.multiply(2)).toEqual(5 * 2 * 2);
      });
    });
  });
});

File: unit_04/w10_d1/lectures/calculatorChall/ruby/spec/calculator_spec.rb
require "./calculator"

RSpec.describe Calculator do
  before(:each) do
    @calculator = Calculator.new
  end

  describe "#add" do
    it "adds two numbers" do
      actual = @calculator.add(2, 3)
      expectation = 5
      expect(actual).to eq(expectation)
    end
  end

  describe "#subtract" do
    it "subtracts two numbers" do
      actual = @calculator.subtract(2, 3)
      expectation = -1
      expect(actual).to eq(expectation)
    end
  end

  describe "#multiply" do
    it "multiplys two numbers" do
      actual = @calculator.multiply(2, 3)
      expectation = 6
      expect(actual).to eq(expectation)
    end
  end

  describe "#divide" do
    it "divides two numbers" do
      actual = @calculator.divide(6, 3)
      expectation = 2
      expect(actual).to eq(expectation)
    end
  end

  context "with previous answer of 5 found" do
    before(:each) do
      @calculator.add(2, 3)
    end

    describe "#add" do
      it "adds a number to previous answer" do
        expect(@calculator.add(2)).to eq(5 + 2)
        expect(@calculator.add(2)).to eq(5 + 2 + 2)
      end
    end

    describe "#subtract" do
      it "subtracts a number to previous answer" do
        expect(@calculator.subtract(2)).to eq(5 - 2)
        expect(@calculator.subtract(2)).to eq(5 - 2 - 2)
      end
    end

    describe "#multiply" do
      it "multiplys a number to previous answer" do
        expect(@calculator.multiply(2)).to eq(5 * 2)
        expect(@calculator.multiply(2)).to eq(5 * 2 * 2)
      end
    end

    describe "#divide" do
      it "divides a number to previous answer" do
        expect(@calculator.divide(2)).to eq(5 / 2)
        expect(@calculator.divide(2)).to eq(5 / 2 / 2)
      end
    end
  end
end
